### ISSUE TITLE
Making `org.kotools.types.text.NotBlankString` type comparable

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#373]
## Description

We would like to make the `org.kotools.types.text.NotBlankString` type comparable by implementing the `Comparable<NotBlankString>` interface.
The comparison between two instances of this type should be done alphabetically, like for the `String` type.

```kotlin
interface NotBlankString : Comparable<NotBlankString> {
    override fun compareTo(other: NotBlankString): Int
}
```

## Dependencies

This issue is blocked by the following ones:
- [x] #372 

## Checklist

- [ ] Make the type implementing `Comparable`, test the behavior of `compareTo` with Kotlin and its compatibility with Java, update the public API binaries and the unreleased changelog.

**Comment by LVMVRQUXL** [comment]
This issue will be implemented by the issue #372.

**Comment by LVMVRQUXL** [comment]
This issue will be implemented by the issue #372.
